### PR TITLE
fix(tabs): use mergeProps in PropGetter callbacks

### LINKED ISSUE #6835: bug(tabs): use mergeProps in PropGetter callbacks
### Which package has an issue?

@yamada-ui/react

### Package Version

main (current)

### Description

The `PropGetter` callback(s) listed below merge the hook's `...rest` and the getter's `...props` via raw spread, instead of using `mergeProps`. As a result, consumer-supplied `ref`, `className`, `style`, and event handlers passed through the getter are silently overwritten by (or fail to merge with) the hook's own values.

**Affected location(s):**

- `packages/react/src/components/tabs/use-tabs.ts:151-159 — `useTabs.getRootProps``
- `packages/react/src/components/tabs/use-tabs.ts:222-249 — `useTab.getRootProps` (uses `handlerAll` for `onClick` / `onFocus`)`
- `packages/react/src/components/tabs/use-tabs.ts:276-289 — `useTabPanel.getRootProps``

### Link to Reproduction

N/A — code-level bug visible by inspection. Any consumer that supplies `className`, `style`, `ref`, or an event handler through the affected getter exhibits the bug.

### Steps to Reproduce

1. Render the affected component / hook consumer.
2. Pass a custom `className` (or `style`, `ref`, event handler) through the affected getter.
3. Observe that the consumer-supplied value is overwritten or fails to merge with the hook's own value, instead of being properly combined.

### Browser

_No response_

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iPadOS
- [ ] iOS
- [ ] Android

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

Yes

### Additional Information

**Suggested fix**

Replace the spread merge with `mergeProps`. Reference (an analogous callback that has already been migrated in the codebase):

```ts
// packages/react/src/components/carousel/use-carousel.ts (getRootProps — already migrated)
const getRootProps: PropGetter<"section"> = useCallback(
  (props) =>
    mergeProps(
      {
        id,
        "aria-roledescription": "carousel",
        "data-orientation": orientation,
      },
      rest,
      props,
      { onMouseEnter, onMouseLeave }, // append internal handlers as the last arg so they merge with the consumer's
    )(),
  [id, onMouseEnter, onMouseLeave, rest, orientation],
)
```

Part of the broader cleanup of `PropGetter` callbacks across `packages/react` that currently use raw spread merges instead of `mergeProps`.

### AI used

- [ ] I did not use AI to create this issue.
- [x] (If there is no check above) I checked the generated content before submitting.

### PR DESCRIPTION
**Pull request by hirotomoyamada**
## AI used

- [ ] I did not use AI to create this PR.
- [x] (If there is no check above) I checked the generated content before submitting.

## Description

`useTabs` / `useTab` / `useTabPanel` の PropGetter で `mergeProps` を使い、利用者が渡すプロップが正しくマージされるようにしました。

## Current behavior (updates)

`rest` と getter 引数の `props` を素のスプレッドで結合していました。

## New behavior

`mergeProps` により `className`・`style`・`ref`・イベントハンドラがマージされます。

## Is this a breaking change (Yes/No):

No

## Additional Information

Closes #6835

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/react/src/components/tabs/use-tabs.ts [hunk 1]
```diff
@@ -220,21 +217,23 @@ export const useTab = ({ id, disabled, index, ...rest }: UseTabProps) => {
...
+          disabled,
+          role: "tab",
+          tabIndex: selected ? 0 : -1,
+        },
+        props,
+        rest,
+        { ref: mergeRefs(ref, register) },
+        { onClick, onFocus },
+      )(),
     [
```
Comment (claude-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts:240]: **Issue:** Argument order `mergeProps(internal, props, rest, ...)` inverts the original precedence for non-event props in `useTab`'s `getRootProps`. The original spread was `...rest, ...props` (user-supplied `props` via `getTabProps(...)` won over `rest`, which is what `useTab({...rest})` received). After this change, `rest` comes after `props` in `mergeProps`, so for non-event/non-ref/non-className fields (e.g. `disabled`, `tabIndex`, `id`, `aria-*`), `rest` now overrides caller-supplied `props`. This is a behavior regression and is also inconsistent with the sibling `useTabs` and `useTabPanel` getters in the same diff, which use `mergeProps(..., rest, props, ...)`.

**Fix:** Swap to `mergeProps({ id, type, "aria-controls": tabPanelId, "aria-selected": selected, "data-orientation": orientation, disabled, role: "tab", tabIndex: selected ? 0 : -1 }, rest, props, { ref: mergeRefs(ref, register) }, { onClick, onFocus })()` to match the previous `...rest, ...props` precedence and stay consistent with the other two getters.

File: packages/react/src/components/tabs/use-tabs.ts [hunk 2]
```diff
@@ -220,21 +217,23 @@ export const useTab = ({ id, disabled, index, ...rest }: UseTabProps) => {
...
-      disabled,
-      role: "tab",
-      tabIndex: selected ? 0 : -1,
-      ...rest,
-      ...props,
-      ref: mergeRefs(ref, register),
-      onClick: handlerAll(props.onClick, rest.onClick, onClick),
-      onFocus: handlerAll(props.onFocus, rest.onFocus, onFocus),
-    }),
+    ({ ref, ...props } = {}) =>
```
Comment (claude-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts:220]: **Issue:** The changeset describes a behavior fix ("so consumer props merge correctly"), but no test is added to lock in the new prop-merging semantics (event chaining order, className/style merge, ref merging, precedence). Without a regression test, this behavior can silently flip again (as already happened in useTab above).

**Fix:** Add a `use-tabs.test.tsx` (or extend the existing tabs tests) covering: (1) `onClick`/`onFocus` chaining order across internal, `rest`, and consumer `props`; (2) `className`/`style` merge on `getRootProps` / `getTabProps` / `getTabPanelProps`; (3) ref forwarding via `mergeRefs`; (4) precedence of consumer `props` over `rest` for plain keys.

**Review by claude-for-yamada-ui — CHANGES_REQUESTED** [review]
- Correctness: Argument order `mergeProps(internal, props, rest, ...)` inverts the original precedence for non-event props in `useTab`'s `getRootProps`. The original spread was `...rest, ...props` (user-supplied `props` via `getTabProps(...)` won over `rest`, which is what `useTab({...rest})` received). After this change, `rest` comes after `props` in `mergeProps`, so for non-event/non-ref/non-className fields (e.g. `disabled`, `tabIndex`, `id`, `aria-*`), `rest` now overrides caller-supplied `props`. This is a behavior regression and is also inconsistent with the sibling `useTabs` and `useTabPanel` getters in the same diff, which use `mergeProps(..., rest, props, ...)`.
- Test coverage: The changeset describes a behavior fix ("so consumer props merge correctly"), but no test is added to lock in the new prop-merging semantics (event chaining order, className/style merge, ref merging, precedence). Without a regression test, this behavior can silently flip again (as already happened in useTab above).

File: packages/react/src/components/tabs/use-tabs.ts [hunk 3]
```diff
@@ -220,21 +217,23 @@ export const useTab = ({ id, disabled, index, ...rest }: UseTabProps) => {
...
+          type: "button",
+          "aria-controls": tabPanelId,
+          "aria-selected": selected,
+          "data-orientation": orientation,
+          disabled,
+          role: "tab",
+          tabIndex: selected ? 0 : -1,
+        },
+        props,
+        rest,
```
Comment (codex-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts]: **Issue:** In `useTab.getRootProps`, `mergeProps` is called with `props` before `rest`, so non-merged keys from `rest` overwrite keys passed to `getRootProps(...)`. This regresses the previous precedence (`...rest, ...props`) and makes consumer overrides unreliable (for example, `tabIndex`, `type`, `aria-*`, `data-*`).

**Fix:** Swap the argument order to `rest, props` for precedence parity, and keep explicit event composition if you need to preserve the current click/focus call order.

File: packages/react/src/components/tabs/use-tabs.ts [hunk 4]
```diff
@@ -220,21 +217,23 @@ export const useTab = ({ id, disabled, index, ...rest }: UseTabProps) => {
...
+        { ref: mergeRefs(ref, register) },
+        { onClick, onFocus },
+      )(),
```
Comment (codex-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts]: **Issue:** This PR changes core prop-merging behavior in `useTab` but adds no regression test for the changed path. Current `tabs.test.tsx` coverage focuses on rendering/a11y/keyboard flow and does not validate `getRootProps` merge precedence or merged handler behavior.

**Fix:** Add regression tests that directly assert `getRootProps` merge behavior (at minimum: precedence between hook-level props and getter props, and composed `onClick`/`onFocus` execution).

**Review by codex-for-yamada-ui — CHANGES_REQUESTED** [review]
- 正確性: In `useTab.getRootProps`, `mergeProps` is called with `props` before `rest`, so non-merged keys from `rest` overwrite keys passed to `getRootProps(...)`. This regresses the previous precedence (`...rest, ...props`) and makes consumer overrides unreliable (for example, `tabIndex`, `type`, `aria-*`, `data-*`).
- テスト: This PR changes core prop-merging behavior in `useTab` but adds no regression test for the changed path. Current `tabs.test.tsx` coverage focuses on rendering/a11y/keyboard flow and does not validate `getRootProps` merge precedence or merged handler behavior.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 3 above)
Comment (composer-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts]: **Issue:** In the same file, `useTabs` and `useTabPanel` use `mergeProps(…, rest, props, …)` so, for non-special-cased keys, `props` (the `getRootProps` argument) is applied after `rest` and wins in the `mergeProps` default branch. `useTab` uses `…, props, rest, …`, which reverses that precedence. That differs from the pre-change `…rest, ...props` (where `…props` won) and from the other two hooks in this PR. For overlapping keys that use the default merge rule, `…rest` can override `getRootProps(…)` input instead of the other way around, so behavior is not aligned with the other getters or the previous implementation.

**Fix:** Use the same `rest` then `props` ordering as `useTabs` and `useTabPanel`, and preserve the intended `onClick` / `onFocus` call order (previously `handlerAll(props…, rest…, internal…)`) in a way that `mergeProps`’s per-step `handlerAll` does not flip—e.g. merge with `{ mergeEvent: false }` where needed and recompose those handlers with `handlerAll` explicitly, or another pattern that matches both attribute precedence and event order.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 4 above)
Comment (composer-for-yamada-ui) [inline packages/react/src/components/tabs/use-tabs.ts]: **Issue:** The diff only updates implementation and a changeset; it adds no tests. The `PropGetter` merge behavior (hook `rest` vs. `getRootProps` input, and handler chaining) is easy to regress and is the core of the change, but nothing in the change exercises it.

**Fix:** Add tests (e.g. in `packages/react/src/components/tabs/tabs.test.tsx` or a dedicated hook test) that assert precedence and composed handlers for the headless `useTab` (and, if you keep one canonical order, for `getRootProps` + overlapping props) so the fix stays covered.

**Review by composer-for-yamada-ui — CHANGES_REQUESTED** [review]
- 正確性, 一貫性: In the same file, `useTabs` and `useTabPanel` use `mergeProps(…, rest, props, …)` so, for non-special-cased keys, `props` (the `getRootProps` argument) is applied after `rest` and wins in the `mergeProps` default branch. `useTab` uses `…, props, rest, …`, which reverses that precedence. That differs from the pre-change `…rest, ...props` (where `…props` won) and from the other two hooks in this PR. For overlapping keys that use the default merge rule, `…rest` can override `getRootProps(…)` input instead of the other way around, so behavior is not aligned with the other getters or the previous implementation.
- テスト: The diff only updates implementation and a changeset; it adds no tests. The `PropGetter` merge behavior (hook `rest` vs. `getRootProps` input, and handler chaining) is easy to regress and is the core of the change, but nothing in the change exercises it.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 3 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts]: Fixed.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 2 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts:220]: Fixed.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 4 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts]: Fixed.

Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts]: Fixed.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 3 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts]: Fixed.

File: packages/react/src/components/tabs/use-tabs.ts (hunk 1 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/tabs/use-tabs.ts:240]: Fixed.

**Review by claude-for-yamada-ui — APPROVED** [review]

**Review by codex-for-yamada-ui — APPROVED** [review]

**Review by composer-for-yamada-ui — APPROVED** [review]
